resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -39,57 +40,107 @@ function Get-ValidTimeZone {
 
-.PARAMETER TimeZone
-    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+.PARAMETER Id
+    The Id to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
 
 .PARAMETER SetTimeZoneAutomatically
-    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
+    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time zone automatically.
+
+.PARAMETER SetTimeAutomatically
+    Whether to set the time automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
 
 .PARAMETER AdjustForDaylightSaving
     Whether to adjust for daylight saving time. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Adjust for daylight saving time automatically.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
 
     This example sets the time zone to Pacific Standard Time.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Get -Property {}
 
     This example gets the current time settings on the machine.
 #>
 [DscResource()]
 class TimeZone {
     [DscProperty(Key)]
-    [string] $TimeZone
+    [string] $Id
 
     [DscProperty()]
     [nullable[bool]] $SetTimeZoneAutomatically
 
+    [DscProperty()]
+    [nullable[bool]] $SetTimeAutomatically
+
     [DscProperty()]
     [nullable[bool]] $AdjustForDaylightSaving
 
-    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Type'
+    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Start'
+    static hidden [string] $SetTimeAutomaticallyProperty = 'Type'
     static hidden [string] $AdjustForDaylightSavingProperty = 'DynamicDaylightTimeDisabled'
+    static hidden [string] $NtpEnabled = 'NTP'
+    static hidden [string] $NtpDisabled = 'NoSync'
+    static hidden [bool] $SupportsDaylightSavingProperty = $false
 
     TimeZone() {
-        $this.TimeZone = (Get-TimeZone).Id
+        $timeZone = Get-TimeZone
+        $this.Id = $timeZone.Id
+
+        # We set the SupportsDaylightSavingProperty to the value that is supported by the current time zone
+        [TimeZone]::SupportsDaylightSavingProperty = $timeZone.SupportsDaylightSavingTime ? $true : $false
```

Comment:
$timeZone.SupportsDaylightSavingTime is already a boolean so no need to have a ternary operator.